frontend/test/components/LoadingAndErrorWrapper.unit.spec.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import sinon from 'sinon'
+
+import LoadingAndErrorWrapper from 'metabase/components/LoadingAndErrorWrapper'
+
+describe('LoadingAndErrorWrapper', () => {
+
+    describe('Loading', () => {
+        it('should display a loading message if given a true loading prop', () => {
+            const wrapper = shallow(
+                <LoadingAndErrorWrapper loading={true}>
+                </LoadingAndErrorWrapper>
+            )
+
+            expect(wrapper.text()).toMatch(/Loading/)
+        })
+
+        it('should display a given child if loading is false', () => {
+            const Child = () => <div>Hey</div>
+
+            const wrapper = shallow(
+                <LoadingAndErrorWrapper loading={false} error={null}>
+                    { () => <Child /> }
+                </LoadingAndErrorWrapper>
+            )
+            expect(wrapper.find(Child).length).toEqual(1)
+        })
+
+        it('should display a given scen during loading', () => {
+            const Scene = () => <div>Fun load animation</div>
+
+            const wrapper = shallow(
+                <LoadingAndErrorWrapper
+                    loading={true}
+                    error={null}
+                    loadingScenes={[<Scene />]}
+                >
+                </LoadingAndErrorWrapper>
+            )
+
+            expect(wrapper.find(Scene).length).toEqual(1)
+        })
+
+        describe('cycling', () => {
+            let clock
+
+            beforeEach(() => {
+                clock = sinon.useFakeTimers()
```

Comment:
You don't necessarily have to use `sinon` here; jest has native support for using fake timers, see http://facebook.github.io/jest/docs/en/timer-mocks.html#content

I think that the API is very similar so should be an easy migration